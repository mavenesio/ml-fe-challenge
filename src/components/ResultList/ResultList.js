import React, {useCallback} from 'react';
import styled from 'styled-components';
import ResultItem from '../ResultItem/ResultItem';

const ResultListContainer = styled.div`
  border-radius:8px;
  background-color:white;
`;

const ResultList = ({data}) => {
  const renderResults = useCallback( (results) => {
    return (results.map( result => {
        return (
          <ResultItem
            key={result.id}
            id={result.id}
            thumbnail={result.picture}
            currency={result.price.currency}
            price={result.price.amount}
            freeShipping={result.free_shipping}
            title={result.title}
            stateName={result.stateName}
          />
        )
        }
    ))
  }, []);

  return (
      <ResultListContainer>
        { data && renderResults(data) }
      </ResultListContainer>
  );
}

export default ResultList;
