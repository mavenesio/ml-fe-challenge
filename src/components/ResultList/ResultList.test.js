import React from 'react';
import ResultList from './ResultList';


const data = [
  {
    condition: "new",
    free_shipping: true,
    id: "MLA835586176",
    picture: "http://mla-s1-p.mlstatic.com/785544-MLA32445402658_102019-I.jpg",
    price: {currency: "$", amount: 223995, decimals: 0},
    stateName: "Capital Federal",
    title: "iPhone 11 128 Gb Negro 4 Gb Ram",
  },
  {
    condition: "new",
    free_shipping: true,
    id: "MLA830184104",
    picture: "http://mla-s1-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
    price: {currency: "$", amount: 183999.99, decimals: 0},
    stateName: "Buenos Aires",
    title: "iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
  }
] 

describe('Result List', () => {
  it('should render without props', () => {   
    expect(<ResultList data={null}/>).toMatchSnapshot();
  });
  it('should render with props', () => {   
    expect(<ResultList data={data}/>).toMatchSnapshot();
  });
});