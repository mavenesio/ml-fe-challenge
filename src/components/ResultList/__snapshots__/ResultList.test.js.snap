// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result List should render with props 1`] = `
<ResultList
  data={
    Array [
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA835586176",
        "picture": "http://mla-s1-p.mlstatic.com/785544-MLA32445402658_102019-I.jpg",
        "price": Object {
          "amount": 223995,
          "currency": "$",
          "decimals": 0,
        },
        "stateName": "Capital Federal",
        "title": "iPhone 11 128 Gb Negro 4 Gb Ram",
      },
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA830184104",
        "picture": "http://mla-s1-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
        "price": Object {
          "amount": 183999.99,
          "currency": "$",
          "decimals": 0,
        },
        "stateName": "Buenos Aires",
        "title": "iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
      },
    ]
  }
/>
`;

exports[`Result List should render without props 1`] = `
<ResultList
  data={null}
/>
`;
